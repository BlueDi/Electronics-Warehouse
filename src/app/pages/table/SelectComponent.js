import React, { Component } from 'react';
import { withCookies } from 'react-cookie';
import { TableSelection } from '@devexpress/dx-react-grid-material-ui';
import { AddToCart } from '@common/components';

class SelectComponent extends Component {
  render() {
    var { cookies, row, selected, onToggle } = this.props;
    var canRequest = cookies.get('can_request') === 'true';
    var cleanRow = Object.create(row);
    delete cleanRow.image;
    return (
      <React.Fragment>
        {canRequest ? (
          <td>
            <AddToCart items={[cleanRow]} simple />
          </td>
        ) : null}
        <TableSelection.Cell
          row={row}
          selected={selected}
          onToggle={onToggle}
        />
      </React.Fragment>
    );
  }
}

export default withCookies(SelectComponent);
