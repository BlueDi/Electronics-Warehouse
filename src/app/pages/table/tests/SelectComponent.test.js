import React from 'react';
import renderer from 'react-test-renderer';
import SelectComponent from '../SelectComponent';

var item1 = {
  description: 'BMW motor',
  details: { details: 'Super detailed description', manufacturer: 'Multicomp' },
  image:
    'iVBORw0KGgoAAAANSUhEUgAAAG8AAABYCAYAAAD7uql1AAAgAElEQVR4Xu29e5RV5Znu+5vfN6/r\nVmtVFXWjoChABFERVCKgUUAjthFzMZqYxLjT3aP79D456d49zj5nZJzTo0f/0+fsHumT7O6xY/be\n2umklcSEaFRuIiiIF0AUFQVFQSAlRRVVRdW6zut3/phrLsâ€¦',
  last_edit: '2018-11-20T11:46:01.628Z',
  last_price: 89.12,
  location: 'HHD12U',
  reference: '213748227'
};

test('Create a selection button for 1 items', () => {
  const component = renderer.create(<SelectComponent row={item1} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
