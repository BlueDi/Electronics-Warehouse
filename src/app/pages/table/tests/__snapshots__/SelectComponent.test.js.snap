// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a selection button for 1 items 1`] = `
<td
  allCookies={Object {}}
  aria-sort={null}
  className="MuiTableCell-root-3 MuiTableCell-paddingCheckbox-9 TableSelectCell-cell-1"
  cookies={
    Cookies {
      "HAS_DOCUMENT_COOKIE": true,
      "changeListeners": Array [
        [Function],
      ],
      "cookies": Object {},
    }
  }
  style={null}
>
  <span
    className="MuiButtonBase-root-27 MuiIconButton-root-21 MuiPrivateSwitchBase-root-17 MuiCheckbox-root-11 MuiCheckbox-colorSecondary-16 TableSelectCell-checkbox-2"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-26"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        checked={false}
        className="MuiPrivateSwitchBase-input-20"
        data-indeterminate={false}
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="MuiTouchRipple-root-39"
    />
  </span>
</td>
`;
